### PR TITLE
Add solana dex price hrs

### PR DESCRIPTION
**Pull request by thanhlmm**
### Contribution type
Please check the type of contribution this pull request is for:

- [x] New spell(s)
- [ ] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name*
- **Description:** [Detailed description of the new spell(s) and their purpose]
- **Who are the new spell(s) for?** [Internal team or general community]
- **How will the new spell(s) be used downstream?** [Description of downstream usage]
- **Implementation details:** [Information on how the spell(s) are implemented]
- **Test instructions:** [How to test the new spell(s)]
- **Related issue(s):** [Link to related issues, if any]

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** [Description of the changes made]

---

### For bug fixes
If you are fixing a bug, please provide the following information:

- **Description:** [Description of the bug fix]
- **Steps to reproduce:** [How to reproduce the bug]
- **Implementation details:** [Information on how the bug was fixed]
- **Test instructions:** [How to test the fix]
- **Related issue(s):** [Link to related issues, if any]

---

### Additional information
Please provide any additional information that might help us review your pull request:

- [Any additional information]

---

Thank you for your contribution!

### CODE REVIEW

File: models/dex/solana/dex_solana_price_hour.sql [hunk 1]
```diff
@@ -0,0 +1,70 @@
+{{
+  config(
+        schema = 'dex_solana_price_hour',
+        alias = 'dex_solana_price_hour',
```
Comment (Hosuke) [inline models/dex/solana/dex_solana_price_hour.sql:4]:
```suggestion
        schema = 'dex_solana',
        alias = 'price_hour',
```

File: models/dex/solana/dex_solana_price_hour.sql [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    raw_data as (
+        SELECT
+            *
+        FROM
+            {{ ref('dex_solana_trades') }}
+        WHERE 1 = 1
+            {% if is_incremental() %}
+            AND {{ incremental_predicate('block_time') }}
+            {% else %}
+            AND block_time >= DATE('2022-03-10')
```
Comment (Hosuke) [inline models/dex/solana/dex_solana_price_hour.sql:29]: We may use the jinja var `project_start_date` here.

**Review by Hosuke — COMMENTED** [review]
cc. @henrystats, do you think this spell may be related to `dex.prices`?

**Comment by jeff-dude** [comment]
thanks for starting the review @Hosuke 

keep in mind, we have similar spells at the project level, let's be sure to keep them consistent (even if that means modifying the orca ones to match):
https://github.com/duneanalytics/spellbook/blob/main/models/orca_whirlpool/orca_whirlpool_token_prices_hour.sql

File: models/dex/solana/dex_solana_price_hour.sql [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    raw_data as (
+        SELECT
+            *
+        FROM
+            {{ ref('dex_solana_trades') }}
+        WHERE 1 = 1
+            {% if is_incremental() %}
+            AND {{ incremental_predicate('block_time') }}
+            {% else %}
+            AND block_time >= DATE('{{project_start_date}}')
```
Comment (Hosuke) [inline models/dex/solana/dex_solana_price_hour.sql:29]: We may fix the syntax:
```suggestion
            AND block_time >= DATE '{{project_start_date}}'
```

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: models/dex/solana/dex_solana_price_hour.sql [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.hour')],
+        unique_key = ['blockchain', 'contract_address', 'hour'],
+        pre_hook='{{ enforce_join_distribution("PARTITIONED") }}',
+        post_hook='{{ expose_spells(\'["solana"]\',
+                            "project",
+                            "dex",
+                            \'["get_nimbus"]\') }}')
+}}
+
+{ % set project_start_date = '2022-03-10' % } --grabbed min block time from whirlpool_solana.whirlpool_call_swap #}
```
Comment (Hosuke) [inline models/dex/solana/dex_solana_price_hour.sql:18]: Here is a syntax typo:
```suggestion
{% set project_start_date = '2022-03-10' %} --grabbed min block time from whirlpool_solana.whirlpool_call_swap
```

**Comment by thanhlmm** [comment]
Hi there, could you please make this PR merged?

**Comment by jeff-dude** [comment]
@Hosuke is this good to go from your perspective?

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @thanhlmm 

[Sample dex_solana_price_hour](https://dune.com/queries/3828050)

File: models/dex/solana/dex_solana_schema.yml [hunk 5]
```diff
@@ -92,3 +92,32 @@ models:
       - &tx_index
         name: tx_index
         description: "index of the transaction in the block slot"
+  - name: dex_solana_price_hour
```
Comment (Hosuke) [inline models/dex/solana/dex_solana_schema.yml:95]:
```suggestion

  - name: dex_solana_price_hour
```

**Comment by Hosuke** [comment]
> @Hosuke is this good to go from your perspective?

The spell alone is good to go.

However, the problem left is whether we need to integrate this spell into `dex.prices`?
https://github.com/duneanalytics/spellbook/blob/main/models/dex/dex_prices.sql

As they have similar (hourly price) schema, but different approach/implementation.
Also `dex.prices` does not cover solana currently.

**Comment by jeff-dude** [comment]
> > @Hosuke is this good to go from your perspective?
> 
> The spell alone is good to go.
> 
> However, the problem left is whether we need to integrate this spell into `dex.prices`? https://github.com/duneanalytics/spellbook/blob/main/models/dex/dex_prices.sql
> 
> As they have similar (hourly price) schema, but different approach/implementation. Also `dex.prices` does not cover solana currently.



> > @Hosuke is this good to go from your perspective?
> 
> The spell alone is good to go.
> 
> However, the problem left is whether we need to integrate this spell into `dex.prices`? https://github.com/duneanalytics/spellbook/blob/main/models/dex/dex_prices.sql
> 
> As they have similar (hourly price) schema, but different approach/implementation. Also `dex.prices` does not cover solana currently.

no, we will leave solana out for now.

**Review by jeff-dude — APPROVED** [review]

**Comment by jeff-dude** [comment]
fyi @aalan3 another solana spell to rebase into your PR

**Pull request merged**